### PR TITLE
 Refactor misc. statement tests for parsers

### PR DESCRIPTION
**Pull request by jugglinmike**
This is in service of gh-1356.

### CODE REVIEW

**Comment by jugglinmike** [comment]
@isiahmeadows I didn't touch a handful of tests from your report in gh-1356:

- `test/language/statements/do-while/S12.6.1_A5.js`
- `test/language/statements/do-while/S12.6.1_A8.js`
- `test/language/statements/for-in/S12.6.4_A3.1.js`
- `test/language/statements/for-in/S12.6.4_A3.js`
- `test/language/statements/for-in/S12.6.4_A4.1.js`
- `test/language/statements/for-in/S12.6.4_A4.js`
- `test/language/statements/while/S12.6.2_A8.js`

These tests concern statement completion values, so their use of `eval` is unavoidable. I was tempted to create corresponding syntax tests, but since they don't describe particularly interesting edge-cases, I decided to skip that.

**Comment by isiahmeadows** [comment]
Okay. FWIW, I didn't even really create 99% of those tests - I just wrapped
them such that they could be used in other places.

File: test/language/statements/break/line-terminators.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    Checking by using eval, inserting LineTerminator between break and
+    Identifier
+---*/
+
+FOR1 : for(var i=1;i<2;i++){
+  LABEL1 : do {
+    break
+FOR1;
+  } while(0);
+}
```
Comment (leobalter) [inline test/language/statements/break/line-terminators.js]: JS can be so beautiful!

File: test/language/statements/for/head-init-expr-check-empty-inc-empty-completion.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+/*---
+info: |
+    The result of evaluating "for( ExpNoIn;Exp;Exp)" loop is returning
+    (normal, evalValue, empty)
+es5id: 12.6.3_A9.1
+description: Using eval
+---*/
+
+var supreme, count;
```
Comment (leobalter) [inline test/language/statements/for/head-init-expr-check-empty-inc-empty-completion.js]: The supreme count of JS.

**Pull request merged**